File: film-information-protal-web/src/main/webapp/js/service/userService.js
//服务层
app.service('userService',function($http){

	//分页 
	this.findPage=function(page,rows){
		return $http.get('../user/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../user/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../user/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../user/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../user/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../user/search.do?page='+page+"&rows="+rows, searchEntity);
	}
	//找回密码
	this.SetCode=function(email){
		return $http.get('../user/SetCode.do?email='+email);
	}

	this.verification=function(code,entity){
		console.log(code);
		console.log(entity);
		return $http.post('../user/verification.do?Code='+code,entity);
	}
});

File: film-information-service-interface/src/main/java/com/michael/service/UserService.java
package com.michael.service;

import com.michael.pojo.User;
import entity.PageResult;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface UserService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<User> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(User user);
	
	
	/**
	 * 修改
	 */
	public void update(User user);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public User findOne(Long id);


	/**
	 * 根据email获取实体
	 * @param userName
	 * @return
	 */
	public User findOne(String userName);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(User user, int pageNum, int pageSize);

	/**
	 * 根据email获取实体
	 * @param user
	 * @return
	 */
	public User findOne(User user);


	/**
	 *
	 * @return
	 */
	public Long countNumber();


}
